File: src/SPS.AdminWeb/src/types/product.ts
// 產品相關類型定義
// 產品相關類型定義
import type {PictureResponse} from './picture';

/**
 * 長度單位枚舉
 */
export const LengthUnit = {
  Millimeter:0,
  Centimeter:1,
  Meter:2,
  Inch:3,
  Foot:4,
} as const ;
export type LengthUnit = (typeof LengthUnit)[keyof typeof LengthUnit];
/**
 * 重量單位枚舉
 */
export const WeightUnit ={
  Gram:0,
  Kilogramz:1,
  Ton:2,
  Pound:3,
  Ounce:4,
} as const;
export type WeightUnit = (typeof WeightUnit)[keyof typeof WeightUnit];

/**
 * 產品文件回應
 */
export interface ProductFileResponse {
  id: string;
  fileNumber: string;
  originalFileName: string;
  fileExtension: string;
  contentType: string;
  fileSize: number;
  formattedFileSize: string;
  fileUrl?: string;
  createdTime: string;
}

/**
 * 產品回應資料
 */
export interface ProductResponse {
  id: number;
  number: string;
  name: string;
  modelNo?: string;
  mixed: boolean;
  unit?: string;
  lengthUnit?: LengthUnit;
  height?: number;
  width?: number;
  depth?: number;
  weightUnit?: WeightUnit;
  netWeight?: number;
  grossWeight?: number;
  conditionedWeight?: number;
  introduction?: string;
  remark?: string;
  categoryId?: number;
  companyId?: string;
  companyName?: string;
  published: boolean;
  /** 封面圖片 */
  cover?: PictureResponse;
  /** 產品圖片列表 */
  pictures: PictureResponse[];
  /** 產品圖片數量 */
  pictureCount: number;
  /** 產品文件列表 */
  files: ProductFileResponse[];
  /** 產品文件數量 */
  fileCount: number;
  createdTime: string;
  updatedTime?: string;
}

/**
 * 產品列表項回應
 */
export interface ProductListItemResponse extends ProductResponse {}

/**
 * 創建產品請求
 */
export interface CreateProductRequest {
  name: string;
  modelNo?: string;
  mixed: boolean;
  unit?: string;
  lengthUnit?: LengthUnit;
  height?: number;
  width?: number;
  depth?: number;
  weightUnit?: WeightUnit;
  netWeight?: number;
  grossWeight?: number;
  conditionedWeight?: number;
  introduction?: string;
  remark?: string;
  categoryId?: number;
  companyId?: string;
  published: boolean;
  /** 封面圖片 ID */
  coverId?: number;
  /** 產品圖片 ID 列表 */
  pictureIds?: number[];
  /** 產品文件 ID 列表（UploadedFile Guid） */
  fileIds?: string[];
}

/**
 * 更新產品請求
 */
export interface UpdateProductRequest {
  name?: string;
  modelNo?: string;
  mixed?: boolean;
  unit?: string;
  lengthUnit?: LengthUnit;
  height?: number;
  width?: number;
  depth?: number;
  weightUnit?: WeightUnit;
  netWeight?: number;
  grossWeight?: number;
  conditionedWeight?: number;
  introduction?: string;
  remark?: string;
  categoryId?: number;
  published?: boolean;
  /** 封面圖片 ID */
  coverId?: number;
  /** 是否移除封面圖片 */
  removeCover?: boolean;
  /** 產品圖片 ID 列表（會取代現有圖片） */
  pictureIds?: number[];
  /** 產品文件 ID 列表（會取代現有文件，UploadedFile Guid） */
  fileIds?: string[];
}

/**
 * 產品查詢參數
 */
export interface ProductQueryParameters {
  search?: string;
  page?: number;
  pageSize?: number;
  sortBy?: string;
  descending?: boolean;
  companyId?: string;
  categoryId?: number;
  published?: boolean;
}

File: src/SPS.Domain/Entities/Product.cs
using SPS.Domain.Common;
using SPS.Domain.Enums;

namespace SPS.Domain.Entities;

public class Product : BaseEntity<int>
{
    public DataMode DataMode { get; set; }
    public string Number { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? ModelNo { get; set; }
    public bool Mixed { get; set; }
    public string? Unit { get; set; }
    public LengthUnit? LengthUnit { get; set; }
    public float? Height { get; set; }
    public float? Width { get; set; }
    public float? Depth { get; set; }
    public WeightUnit? WeightUnit { get; set; }
    public double? NetWeight { get; set; }
    public double? GrossWeight { get; set; }
    public double? ConditionedWeight { get; set; }
    public string? Introduction { get; set; }
    public int? DescriptionId { get; set; }
    public string? Remark { get; set; }
    public int? CoverId { get; set; }
    public int? BaseId { get; set; }
    public int? CategoryId { get; set; }
    public Guid? CompanyId { get; set; }
    public bool Published { get; set; }

    // Navigation properties
    public StringResource? Description { get; set; }
    public Picture? Cover { get; set; }
    public Product? Base { get; set; }
    public Category? Category { get; set; }
    public Company? Company { get; set; }
    public ICollection<Product> Variants { get; set; } = new List<Product>();
    public ICollection<Picture> Pictures { get; set; } = new List<Picture>();
    public ICollection<File> Files { get; set; } = new List<File>();
    public ICollection<UploadedFile> UploadedFiles { get; set; } = new List<UploadedFile>();
}
